hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/ImmediateExecutorService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.fixtures;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An {@link ExecutorService} that executes tasks immediately in the calling thread, rather than in a separate thread.
+ * This is useful for synchronous testing where the API involves an {@link ExecutorService}.
+ */
+public class ImmediateExecutorService extends AbstractExecutorService {
+
+    private boolean shutdown = false;
+
+    @Override
+    public synchronized void shutdown() {
+        shutdown = true;
+    }
+
+    @Override
+    public synchronized List<Runnable> shutdownNow() {
+        // Note, since this method is synchronized, it is not possible for another thread to concurrently add a new
```

Comment:
There is no queue to be modified concurrently, so likely this method (and all the others) doesn't need to be synchronized at all (particularly if the shutdown boolean on line 33 is declared volatile).